docs/user/alerting/action-variables.asciidoc

```diff
@@ -125,3 +129,193 @@ triggering data was:
+- de-dent support for leading whitespace in multi-line strings
+- also supports legal JSON documents
+
+To use it, surround your Hjson content with `{{#ParseHjson}}your content here{{/ParseHjson}}`.
+
+For example, here's an example template:
+ 
+[source]
+----
+{{#ParseHjson}}
+{
+  # add the rule id and name to the JSON document
+  ruleId:   "{{rule.id}}"
+  ruleName: "{{rule.name}}"
+}
+{{/ParseHjson}}
+----
+
+When rendered, this will generate:
+ 
+[source,json]
+----
+    {
+      "ruleId": "<the rule id is here>",
+      "ruleName": "<the rule name is here>"
+    }
+----
+
+[discrete]
+===== FormatDate
+
+The FormatDate lambda provides date formatting capabilities.  Dates can be formatted in an arbitrary time zone and with an arbitrary format string.
+
+To use it, surround the date and formatting parameters with `{{#FormatDate}}<date and formatting parameters>{{/FormatDate}}`.
+
+The format of the text passed to the lambda is: `<date>; <time zone>; <date format>`, where semicolons (`;`) separate each parameter.  The `<date>` parameter is required, and the `<time zone>` and `<date format>` parameters are optional.  The default time zone if not provided is `"UTC"`, and the default date format is `"YYYY-MM-DD hh:mma"`.  For example, the following templates all render the same value:
+ 
+[source]
+----
+    {{#FormatDate}} {{{timestamp}}} {{/FormatDate}}
+    {{#FormatDate}} {{{timestamp}}} ; UTC {{/FormatDate}}
+    {{#FormatDate}} {{{timestamp}}} ; UTC; YYYY-MM-DD hh:mma {{/FormatDate}}
+    {{#FormatDate}} {{{timestamp}}} ; ; YYYY-MM-DD hh:mma {{/FormatDate}}
+----
+
+The `<time zone>` parameter must be a valid time zone identifier as listed in https://en.wikipedia.org/wiki/List_of_tz_database_time_zones[TZ database time zone names], such as `"America/New_York"`.
+
+The `<date format>` parameter must be a valid date format string as described in the https://momentjs.com/docs/#/displaying/[Moment `format()` documentation].  For example, the date format  `"YYYY-MM-DD hh:mma"` will render in the following format: `"2023-04-24 11:21pm"`.
+
+The date value itself should usually be referenced with triple braces, as some characters in date strings may contain values which are escaped, which would prevent them from being parsed as dates.
+
+[discrete]
+[[mustache-examples]]
+=== Mustache examples
+
+Below is a Mustache template for an email action sent from an elasticsearch query rule.
+
+[source]
+----
+# {{context.title}}
+
+{{#FormatDate}} {{{date}}} ; America/New_York {{/FormatDate}}
+
+{{context.conditions}}
+
+**documents** _[view in Discover]({{{context.link}}})_
+
+| provider | action | duration |
+| -------- | ------ | -------- |
+{{#context.hits}}{{#_source.event}}| {{provider}} | {{action}} | {{#duration}}{{#EvalMath}} round( {{{duration}}} / 1000 / 1000 / 1000 ) {{/EvalMath}} sec{{/duration}} {{^duration}}-n/a-{{/duration}} |{{/_source.event}}
+{{/context.hits}}
+----
+
```

Comment:
To use callouts, you'd put the explanation directly after the codeblock like this:
```suggestion
<1> Renders the value of the `context.title` variable as a level 1 heading.
```